x/cdp/types/genesis.go

```diff
@@ -55,9 +56,11 @@ func (gs GenesisState) Validate() error {
 		return fmt.Errorf("previous distribution time not set")
 	}
 
-	if gs.DebtDenom == "" {
+	if strings.TrimSpace(gs.DebtDenom) == "" {
 		return fmt.Errorf("debt denom not set")
-
+	}
+	if strings.TrimSpace(gs.GovDenom) == "" {
+		return fmt.Errorf("gov denom not set")
 	}
```

Comment:
sdk.Coin's `ValidateDenom()` checks against regex `[a-z][a-z0-9]{2,15}`, so example string "-!))" will pass this check but panic later on sdk.Coins' `AmountOf()` since the method requires a valid denom. Either validate against the same regex or use `err := myCoin.ValidateDenom(strings.TrimSpace(gs.GovDenom))` and check that `err` is nil.